lib/record_collection.rb

```diff
@@ -0,0 +1,25 @@
+module RegistersClient
+  class RecordCollection
+    include Enumerable
+
+    def initialize(data, page_size = 100)
+      @data = data
+      @page_size = page_size
+    end
+
+    def each
+      @data.each do |record|
+        yield record
+      end
+    end
```

Comment:
I'm thinking that the `#each` method could be implemented with

```ruby
    def each(&block)
      @data.each(&block)
    end
```

Similar result tbh, use it or not at your discretion :)

Reply:
Yeah I did see that :)